lib/services/systemd.c

```diff
@@ -136,6 +139,50 @@ systemd_call_simple_method(const char *method)
     return reply;
 }
 
+/*!
+ * \internal
+ * \brief Subscribe to D-Bus signals from systemd
+ *
+ * Systemd does not broadcast signal messages unless at least one client has
+ * called the \c Subscribe() method. Also, a D-Bus client ignores broadcast
+ * messages unless an appropriate match rule is set, so we set one here.
+ *
+ * \return Standard Pacemaker return code
+ */
+static int
+subscribe_to_signals(void)
+{
+    DBusMessage *reply = NULL;
+    DBusError error;
+
+    /* Tell D-Bus to accept signal messages from systemd.
+     * https://dbus.freedesktop.org/doc/dbus-specification.html#message-bus-routing-match-rules
+     */
+    dbus_error_init(&error);
+    dbus_bus_add_match(systemd_proxy,
+                       "type='signal',"
+                       "sender='" BUS_NAME "',"
+                       "interface='" BUS_NAME_MANAGER "',"
+                       "path='" BUS_PATH "'",
+                       &error);
+
+    if (dbus_error_is_set(&error)) {
+        crm_err("Could not listen for systemd DBus signals: %s " QB_XS " (%s)",
+                error.message, error.name);
+        dbus_error_free(&error);
+        return ECOMM;
+    }
+
+    // Tell systemd to broadcast signals
+    reply = systemd_call_simple_method("Subscribe");
+    if (reply == NULL) {
+        return ECOMM;
```

Comment:
This is probably unnecessary, but it might be nice to call `dbus_bus_remove_match` here to help clean up what we initialized.

Reply:
Thanks! It's a good idea as a best practice, to sort of "close the loop" and make the function itself as correct as possible. So I'll plan to add that call. In practice, it won't matter: the caller will free the connection, which frees the match rule.

Reply:
Done in latest push